File: components/common/lib/ProtobufCANInterface/test/test_can_interface.py
# for testing the compiled c++ library

import sys
import os

# Add the build directory to the Python path
sys.path.append(os.path.join(os.path.dirname(__file__), 'build'))

# Import the library
from ProtobufCANInterface import ProtobufCANInterface

# Create an instance of the library
can_interface = ProtobufCANInterface(0x01)

# Initialize the CAN interface
can_interface.begin(500000)

# Register a handler for the lights component
def lights_handler(message_type, component_type, component_id, command_id, value_type, value):
    print(f"Received lights message: {message_type}, {component_type}, {component_id}, {command_id}, {value_type}, {value}")

can_interface.register_handler(0x01, 0x01, 0x01, lights_handler)

# Send a message to the lights component
can_interface.send_message(0x01, 0x01, 0x01, 0x01, 0x01, 0x01)

# Process incoming messages
can_interface.process()

File: components/common/lib/ProtobufCANInterface/test/test_can_interface.cpp
#include "ProtobufCANInterface.h"
#include <iostream>

int main() {
    std::cout << "Testing ProtobufCANInterface in non-Arduino environment" << std::endl;
    
    // Create a CAN interface instance with node ID 1
    ProtobufCANInterface can(1);
    
    // Initialize with default baud rate
    can.begin();
    
    // Send a test message
    bool result = can.sendMessage(
        kart_common_MessageType_COMMAND,
        kart_common_ComponentType_LIGHTS,
        1,  // component_id (e.g., front lights)
        0,  // command_id (e.g., mode)
        kart_common_ValueType_UINT8,
        1   // value (e.g., ON)
    );
    
    std::cout << "Message send result: " << (result ? "Success" : "Failed") << std::endl;
    
    // Process any incoming messages (would be a no-op in this environment)
    can.process();
    
    return 0;
} 